[release][codesign] make script python3 compatible

XilaiZhang opened the pull request:
The goal is to have every new bot run with python3 for better scalability, given that python2 has been deprecated. A non deterministic versioning of python could cause confusions among users.

XilaiZhang committed:
python3 compatible

XilaiZhang commented:
umm seems like I don't have permissions to add reviewers. would be great if @christopherfujino @jmagman could take a look. Thank you!

christopherfujino commented on codesign.py:
```diff
@@ -395,7 +395,8 @@ def run_and_return_output(command):
         stderr=subprocess.PIPE)
-    return proc.stdout.readlines() + proc.stderr.readlines()
+    byteList = proc.stdout.readlines() + proc.stderr.readlines()
+    return [x.decode('utf-8') for x in byteList]
```
what if a unicode character is more than one byte long?

christopherfujino replied:
Will passing `text=True` do what we want? https://docs.python.org/3/library/subprocess.html#popen-constructor

christopherfujino replied:
sounds like it: https://docs.python.org/3/library/subprocess.html#frequently-used-arguments

XilaiZhang replied:
Thanks for the reply! that's true, I will incorporate this change in the next commit!

XilaiZhang committed:
notary tool for version 13

XilaiZhang commented:
Newest commit added support for notarytool based on xcode version. Tested locally and remotely. I will update the newly required settings in doc as well.

christopherfujino commented on codesign.py:
```diff
@@ -757,6 +795,10 @@ def main(args, bucket_prefix):
 CODESIGN_USERNAME = os.environ['CODESIGN_USERNAME']
```
can we delete this?

XilaiZhang replied:
umm This seem to be used in a lot of places. Maybe I am missing something, why would we like to delete it?

christopherfujino commented on codesign.py:
```diff
@@ -438,9 +439,38 @@ def upload_zip_to_notary(archive_path):
+            'notarytool',
+            'submit',
+            archive_path,
+            '--wait',
```
now that we're waiting, does this take longer than the previous version?

XilaiZhang committed:
notarytool status update

XilaiZhang commented:
Added notarytool support for polling process as well. removed three armv7 files (context https://github.com/flutter/engine/pull/32664/files#diff-a42df9dcaea2b4e4074966cfe07bb92ee27d8972538d58434017c0c937107679R66) I timed multiple runs from start to finish. The time it takes to success is actually short, around 5 minutes.
```
Codesigning & Notarization successful
Success
./codesign.py b41b4de7d6457899f65ac9267cf7f4d0c57bb04f  119.30s user 23.35s system 49% cpu 4:49.08 total
```
 However, I have only observed one status returned from `notary info`, and that is `Accepted`. I have left comment in code regarding this suspicious status.

XilaiZhang commented on codesign.py:
```diff
@@ -474,41 +505,67 @@ def upload_zip_to_notary(archive_path):
-        log('Notarization is still pending...\n')
-        return False
+    if isNotaryTool:
+        if status == 'Accepted': #seems like accpeted is the equivalent of success, if it means in progress it never finishes
```
seems like accpeted is the equivalent of success, if it means in progress it never finishes

christopherfujino replied:
can manually verify these with `spctl`? https://stackoverflow.com/questions/55074168/how-do-i-verify-that-a-macos-pkg-is-notarized#:~:text=The%20notarization%20process%20works%20by,for%20it%20to%20be%20notarized.

XilaiZhang replied:
synced with Chris in microkitchen, was caused by some caching issues and was resolved.

XilaiZhang commented:
update: I have retrieved the different statuses from notary tool. I am currently working on some errors observed :`The signature does not include a secure timestamp`, `The executable does not have the hardened runtime enabled`, `The binary is not signed` etc.

XilaiZhang committed:
add missing binaries, tuning

XilaiZhang commented:
just finished testing locally and remotely. The manual code signing should now be complete.

christopherfujino commented:
> just finished testing locally and remotely. The manual code signing should now be complete.

nice, I will try this out today

christopherfujino reviewed (approved):
LGTM

christopherfujino reviewed (changes requested):
Ok, per discussion offline maybe this isn't hte version of the script being run in production right now

jmagman commented:
@christopherfujino this can probably be closed?

christopherfujino commented:
Yep, this script is deprecated
